### DISCUSSION TITLE
Debian build help is needed

### DISCUSSION THREAD

**Discussion by herasymenko-bohdan** [garsipal/valhalla-project#70] [Help] [answered]
The linux build section on the main page says:
> In order to build the source code by using the Makefile included in the source directory, you need to ensure you have the SDL2, SDL2-image, SDL2-mixer and OpenGL development libraries installed.

Can someone please elaborate on how to install those libraries? And where to install them from?

If windows and mac can have their builds provided, can an app image be provided for the GNU?

**Comment by garsipal** [comment, marked as answer]
Installing these development libraries on Debian is quite straightforward. Open a terminal and do:

```
sudo apt update
sudo apt-get install libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev 
```
This should automatically install the SDL2 dev libraries and GL dependencies.

To manually install the SDL2 libraries (without package managers) you'll need to roughly follow these steps:
- Get a copy of the SDL2 source code;
- Run the configure script;
- Do `make` to compile the SDL2 source code.
- Do `make install` to install the SDL build on the system.

More details [here](https://wiki.libsdl.org/SDL2/Installation).

<hr>

I'll look into adding more details/links to the build instructions to make things more accessible, especially since such questions are understandable given that there isn't a proper way of distributing the project on GNU/Linux yet.  

The next small release will address some issues from the previous version, bring in a few new features and include precompiled Ubuntu binaries, but a next major release would need to plan out a smooth distribution strategy.

**Comment by herasymenko-bohdan** [comment]
Can confirm, the `apt install` did work, but the binaries mentioned here were not created:
> This will also install the binaries and make them accessible in the bin/bin_unix/ directory

Launching the game via the `valhalla_unix` file in the root folder did attempt to start the game, but presented the `Tesseract fatal error: Could not find core textures` error popup. The terminal window hinted that the textures had to be placed under the `/home/user/.valhallaproject` folder. Having downloaded the [data repo](https://github.com/project-valhalla/data) and placed in there under the `data` folder, so that the path `/home/user/.valhallaproject/data/readme.txt` became correct. All this seems to have worked.
I think that install tutorial should mention these steps
